onnxruntime/contrib_ops/cuda/math/bias_dropout_impl.cu

```diff
@@ -49,34 +49,82 @@ __global__ void BiasDropoutKernel(
   curandStatePhilox4_32_10_t state;
   curand_init(seeds.first, idx, seeds.second, &state);
 
-  // We ensure every thread generates the same number of random numbers (by rounding
-  // up the size) and at the same timestep (by syncing threads).
-  // From CUDA curand documentation:
-  //   The Philox_4x32_10 algorithm is closely tied to the thread and block count.
-  //   Each thread computes 4 random numbers in the same time thus the most efficient
-  //   use of Philox_4x32_10 is to generate a multiple of 4 times number of threads.
-  for (CUDA_LONG id = idx; id < rounded_size; id += step_size) {
-    float4 rand = curand_uniform4(&state);
-
-  #pragma unroll
-    for (CUDA_LONG i = 0; i < UNROLL; i++) {
-      CUDA_LONG li = id + gridDim.x * blockDim.x * i;
-      if (li < N) {
-        int offset = fdm_dim.mod(li);
+  float4 rand;
+
+  if (N % UNROLL != 0) { 
```

Comment:
I would suggest writing a separate Kernel for the vectorized impl. 
The selection of the kernel can be done at kernel launching time.

Reply:
Same comments applies to Dropout and DropoutGrad kernels.